extensions/wikia/Helios/User.class.php

```diff
@@ -27,35 +29,57 @@ public static function debugLogin( $password, $callingMethod )
 	}
 
 	/**
-	 * Creates a MediaWiki User object based on the token given in the HTTP request.
+	 * Extracts access token from HTTP request data.
+	 *
+	 * @param \WebRequest $request the HTTP request data as an object
+	 * @return String access token
 	 */
-	public static function newFromToken( \WebRequest $request )
-	{
+	public static function getAccessToken( \WebRequest $request ) {
+		// A cookie takes precedence over an HTTP header.
+		$token = $request->getCookie( self::ACCESS_TOKEN_COOKIE, '' );
 
-		$header = $request->getHeader( 'AUTHORIZATION' );
-
-		if ( $header ) {
+		// No access token in the cookie, try the HTTP header.
+		if ( ! $token ) {
+			$header = $request->getHeader( 'AUTHORIZATION' );
 
 			$matches = [];
 			preg_match( '/^Bearer\s*(\S*)$/', $header, $matches );
 
-			if ( !empty( $matches[1] ) ) {
-
-				global $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret;
+			if ( ! empty( $matches[1] ) ) {
+				$token = $matches[1];
+			}
+		}
 
-				$heliosClient = new Client( $wgHeliosBaseUri, $wgHeliosClientId, $wgHeliosClientSecret );
+		return $token;
+	}
 
-				try {
-					// Authenticate with the token and create a MediaWiki User object.
-					$tokenInfo = $heliosClient->info( $matches[1] );
-					if ( !empty( $tokenInfo->user_id ) ) {
-						return \User::newFromId( $tokenInfo->user_id );
-					}
+	/**
+	 * Creates a MediaWiki User object based on the token given in the HTTP request.
+	 *
+	 * @param \WebRequest $request the HTTP request data as an object
+	 *
+	 * @return \User on successful authentication
+	 */
+	public static function newFromToken( \WebRequest $request )
```

Comment:
there is a bunch of `newFrom` and `initFrom` methods that could be renamed to `newFromWebRequest`, too, namely `newFromSession`, `newFromConfirmationCode`, `initFromCookie`, etc. I'll leave it as it is.